src/ast/passes/codegen_llvm.cpp

```diff
@@ -635,12 +636,16 @@ ScopedExpr CodegenLLVM::kstack_ustack(const std::string &ident,
   // bpf_get_stackid() is kind of broken by design and can suffer from hash
   // collisions.
   // More details here: https://github.com/bpftrace/bpftrace/issues/2962
-  Value *murmur_hash_2 = b_.CreateCall(
-      murmur_hash_2_func_,
-      { stack_trace, trunc_nr_stack_frames, seed },
-      "murmur_hash_2");
-
-  b_.CreateStore(murmur_hash_2,
+  std::array<llvm::Type *, 3> arg_types = { b_.getPtrTy(),
+                                            b_.getInt8Ty(),
+                                            b_.getInt64Ty() };
+  std::array<llvm::Value *, 3> arg_values = { stack_trace,
+                                              trunc_nr_stack_frames,
+                                              seed };
+  auto murmur_hash_2 = createExternFuncCall(
```

Comment:
I don't object to the refactoring here, but you're injecting implicit arguments and such, which should be part of a lowering pass ahead of codegen. We can chat about this offline also.